controllers/idler/idler_controller.go

```diff
@@ -132,7 +132,13 @@ func (r *Reconciler) ensureIdling(logger logr.Logger, idler *toolchainv1alpha1.I
 			// Already tracking this pod. Check the timeout.
 			if time.Now().After(trackedPod.StartTime.Add(time.Duration(idler.Spec.TimeoutSeconds) * time.Second)) {
 				podLogger.Info("Pod running for too long. Killing the pod.", "start_time", trackedPod.StartTime.Format("2006-01-02T15:04:05Z"), "timeout_seconds", idler.Spec.TimeoutSeconds)
-
+				var podreason string
+				podcondition := pod.Status.Conditions
+				if podcondition != nil {
+					podreason = podcondition[0].Reason
```

Comment:
Shouldn't we check the status of a specific condition? I guess the "Ready" condition?

Reply:
All the condition "type" will have a _reason_ as "Pod Completed" , so i didnt check that .. but can add to check the reason only for "ready" condition type